examples/toy_mixture_model.py

```diff
@@ -0,0 +1,130 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+ A toy mixture model to provide a simple example for implementing discrete enumeration.
+
+ (A) -> [B] -> (C)
+
+ A is an observed Bernoulli variable with Beta prior.
+ B is a hidden variable which is a mixture of two Bernoulli distributions (with Beta priors),
+ chosen by A being true or false.
+ C is observed, and like B, is a mixture of two Bernoulli distributions (with Beta priors),
+ chosen by B being true or false.
+ There is a plate over the three variables for n independent observations of data.
+
+ Because B is hidden and discrete we wish to marginalize it out of the model.
+ This is done by:
+    1) marking the model method with `@pyro.infer.config_enumerate`
+    2) marking the B sample site in the model with `infer={"enumerate": "parallel"}`
+    3) passing `pyro.infer.SVI` the `pyro.infer.TraceEnum_ELBO` loss function
+"""
+
+import os
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from torch.distributions import constraints
+from torch.distributions.bernoulli import Bernoulli
+from torch.distributions.beta import Beta
+from tqdm import tqdm
+import pyro
+import pyro.distributions as dist
+import pyro.infer
+import pyro.optim
+from pyro.ops.indexing import Vindex
+
+smoke_test = ('CI' in os.environ)
```

Comment:
We use the environment variable idiom `'CI' in os.environ` idiom only for notebook tutorials where it is difficult to pass command line arguments. For script tutorials we instead use `argparse` and pass options to the script test in `test_examples.py`. Could you remove the `smoke_test = ...` and instead add a little parser with a version check
```py
if __name__ == "__main__":
    assert pyro.__version__.startswith('1.3.1')
    parser = argparse.ArgumentParser(description="Toy mixture model")
    parser.add_argument("-n", "--num-steps", default=4000, type=int)
    args = parser.parse_args()
    main(args)
```
Then add plumbing of `args` through to `train()`. Then in `test_examples.py` add `--num-samples=1` to the invocation (as I've commented below).

Reply:
Done.

Reply:
Great, can you remove this `smoke_test = ...` line since it is no longer used?

Reply:
Whoops, done.